### PR TITLE
manage proxy-client / aggregator certs

### LINKED ISSUE #1404: Should set the RequestHeader Client CA options
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

Minikube should generate and pass in a RequestHeader client CA (in addition to the normal client CA) so that the `extension-apiserver-authentication` ConfigMap is autopopulated with the requestheader client CA (see https://github.com/kubernetes/kubernetes/blob/master/pkg/master/client_ca_hook.go).

This is used by custom API servers to trust the authentication information passed to them by the API server aggregator proxy (a separate pod in 1.6, is/will be part of the master for 1.7).  The generic API server code automatically tries to use the information in this map (when the requestheader client CA is not manually passed via a command line option), so having the value populated is useful even when the user is not running the proxy.

cc @pmorie @arschles kubernetes-incubator/service-catalog#737

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1404 

cc @DirectXMan12 @dlorenc @arschles

### CODE REVIEW

**Comment by r2d4** [comment]
@minikube-bot ok to test

**Comment by krancour** [comment]
Test failures don't appear related to the change. Is someone able to restart CI please?

File: pkg/minikube/bootstrapper/certs.go [hunk 1]
```diff
@@ -95,16 +108,38 @@ func SetupCerts(cmd CommandRunner, k8s KubernetesConfig) error {
...
 		if err := util.GenerateCACert(caCert, caKey, name); err != nil {
-			return errors.Wrap(err, "Error generating certificate")
+			return errors.Wrap(err, "Error generating CA certificate")
 		}
 	}
 
 	ips := []net.IP{ip, internalIP}
-	if err := util.GenerateSignedCert(pub, priv, ips, util.GetAlternateDNS(dnsDomain), caCert, caKey); err != nil {
+	if err := util.GenerateSignedCert(
+		pub, priv, "minikube", ips, util.GetAlternateDNS(dnsDomain),
```
Comment (r2d4) [inline pkg/minikube/bootstrapper/certs.go]: Any reason why we parameterized and hardcoded the machine name here?  I think this should be getting its value from KubernetesConfig.NodeName, which is default "minikube" but can be configured.

Comment (krancour) [reply pkg/minikube/bootstrapper/certs.go]: I didn't want the new cert (the proxy-client cert to be used by the aggregator) to ben named "minikube" (or whatever the machine name was), so I updated the function signature of `GenerateSignedCert (...)` to require that you pass in the subject name...

But it's definitely a mistake on my part that when invoking that modified function, I passed in a hardcoded value. I'll fix this!

Comment (krancour) [reply pkg/minikube/bootstrapper/certs.go]: Actually... taking a second look at this, it was always hardcoded as "minikube". All that's happened here is that that hardcoded value is now an argument _to_ `GenerateSignedCert (...)`, where it was previously hardcoded _in_ `GenerateSignedCert(...)`.

Comment (r2d4) [reply pkg/minikube/bootstrapper/certs.go]: Yeah - I actually like that you parameterized it in the function here.  I think it was incorrectly hardcoded in the function before.

Comment (r2d4) [reply pkg/minikube/bootstrapper/certs.go]: We don't have a integration test for this, but we should verify that if you specify the `--profile` flag to minikube start, that the certs get generated correctly and everything comes up fine.

Comment (krancour) [reply pkg/minikube/bootstrapper/certs.go]: I'll verify that.

Comment (krancour) [reply pkg/minikube/bootstrapper/certs.go]: @r2d4 The new feature is working as expected even with the `--profile` flag used. As with the apiserver's serving cert, the new cert is created in `~/.minikube` and is common to all minikube VMs.

**Comment by krancour** [comment]
@r2d4 can we get CI restarted on this PR? The failures don't look related to the changes.

**Comment by r2d4** [comment]
I wouldn't worry about those OSX failures.  

Were you still waiting to fix 
> I didn't want the new cert (the proxy-client cert to be used by the aggregator) to be named "minikube" (or whatever the machine name was), so I updated the function signature of GenerateSignedCert (...) to require that you pass in the subject name...
> But it's definitely a mistake on my part that when invoking that modified function, I passed in a hardcoded value. I'll fix this!

Or is this ready to merge?

**Comment by krancour** [comment]
> Were you still waiting to fix
> 
>> I didn't want the new cert (the proxy-client cert to be used by the aggregator) to be named "minikube" (or whatever the machine name was), so I updated the function signature of GenerateSignedCert (...) to require that you pass in the subject name...
But it's definitely a mistake on my part that when invoking that modified function, I passed in a hardcoded value. I'll fix this!

Nope... if that's ok. As we discussed, that piece was always hard-coded. All that has happened is _where_ it's hardcoded moved. I think if that needs to change (I don't disagree) that should be separate from this PR.

So yes... ready to merge imo.

**Pull request merged**